src/tool/subcommands/shed_cmd/f3.rs

```diff
@@ -0,0 +1,111 @@
+// Copyright 2019-2025 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use crate::{
+    rpc::{
+        self,
+        eth::types::{EthAddress, EthBytes, EthCallMessage},
+        f3::F3Manifest,
+        state::StateCall,
+        RpcMethodExt,
+    },
+    shim::message::Message,
+};
+use clap::Subcommand;
+use flate2::read::DeflateDecoder;
+use std::{io::Read, str::FromStr as _};
+
+const MAX_PAYLOAD_LEN: usize = 4 << 10;
+
+#[derive(Debug, Subcommand)]
+pub enum F3Commands {
+    CheckActivation {
+        /// Contract address
+        #[arg(long, required = true)]
+        contract: String,
```

Comment:
Not a huge deal given it's a testing tool, but it'd be great to use concrete types here instead of `String` if possible.

Reply:
Fixed.